### ISSUE TITLE
How to create and install plugin tgz

### ISSUE THREAD

**Issue by mblaschke** [helm/helm#11272]
Is the plugin installation with tgz still supported/working? It's mentioned in documentation at https://helm.sh/docs/topics/plugins/#installing-a-plugin

But when trying to download a .tgz from github releases this happens:

```
$ helm plugin install https://github.com/xxxx/xxxx/releases/download/xx-xx-xx/helm-plugin.tgz
Error: Unable to get repository: Cloning into '/Users/mblaschke/Library/Caches/helm/plugins/https-github.com-xxxx-helm-xxxxx-releases-download-xx-xx-xx-helm-plugin.tgz'...
remote: Not Found
fatal: repository 'https://github.com/xxxx/xxxx/releases/download/xx-xx-xx/helm-plugin.tgz/' not found
: exit status 128
```

when fetching the tgz and trying it locally it also fails:

```
$ helm --debug plugin install release-assets/helm-plugin.tgz
Error: plugin metadata (plugin.yaml) missing
helm.go:84: [debug] plugin metadata (plugin.yaml) missing
helm.sh/helm/v3/pkg/plugin/installer.init
	helm.sh/helm/v3/pkg/plugin/installer/installer.go:32
runtime.doInit
	runtime/proc.go:6222
runtime.doInit
	runtime/proc.go:6199
runtime.main
	runtime/proc.go:233
runtime.goexit
	runtime/asm_arm64.s:1263


$ tar tzvf release-assets/helm-plugin.tgz | grep plugin
-rw-r--r--  0 mblaschke staff      693 Aug 25 15:39 plugin.yaml


$ file release-assets/helm-plugin.tgz
release-assets/helm-plugin.tgz: gzip compressed data, last modified: Thu Aug 25 13:39:09 2022, from Unix, original size modulo 2^32 73625600
```

might be also related to #10402

Output of `helm version`:
version.BuildInfo{Version:"v3.9.2", GitCommit:"1addefbfe665c350f4daf868a9adc5600cc064fd", GitTreeState:"clean", GoVersion:"go1.18.4"

Output of `kubectl version`:
n/A

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
n/A

**Comment by phil9909** [comment]
Hi @mblaschke,

- `helm plugin install https://github.com/xxxx/xxxx/releases/download/xx-xx-xx/helm-plugin.tgz` seems to be a problem in `isRemoteHTTPArchive` not following redirects.
- `helm --debug plugin install release-assets/helm-plugin.tgz` this will not work because it expects a folder. You need to extract the tar first. I guess the error message could be improved here.
  Try something like:
  ```bash
  mkdir tmp
  tar xf release-assets/helm-plugin.tgz -C tmp
  helm plugin install tmp
  rm -rf tmp
  ```
  I hope this unblocks you,

Do you think you could create Pull Requests to improve this?

**Comment by s7m4b4** [comment]
I had a brief look into this and as far as I can tell the client does follow redirects (Go's http client does this by default) however since Github is using `application/octet-stream` as its media type it fails the `mediaTypeToExtension` check. This is an easy fix and I can raise a PR, however I discovered a few other issues with the HTTPInstaller. In short -

- Install fails if the root dir in the archive is >1 level deep (I tried fixing this but it led to the below)
- Install fails if plugins.yaml is not in the root dir

4/5 random plugins I tried testing using the HTTPInstaller failed even with the fix above. This is partly due to how plugins devs have bundled their plugins, so I'm not sure if the maintainers want to account for this. In any case I can raise a followup issue.

**Comment by hickeyma** [comment]
With PRs #11508 and #11389, we are definitely improving the issues around installing from archives. However, we still have open issues around archives where files are not all in root. 

@s7m4b4 Do you have an ideas around fixing this? If not then maybe a doc PR should be pushed to document the limitation of install plugins from an archive?
